src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && (this[i] == null || this[i] === undefined)) {
+        str += '';
+      } else if (i === this.length - 1) {
+        str += this[i];
+      } else if (this[i] == null || this[i] === undefined) {
+        str += separator;
+      } else {
+        str = str + this[i] + separator;
+      }
```

Comment:
i propose you to improve it
if element !-- undefined or null => add element
if it's not last element => add separator
2 ifs will be enough for this method :)